x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -258,6 +259,17 @@ public void onResponse(Versioned<LogicalPlan> analyzedPlan) {
         );
     }
 
+    private static String projectRouting(EsqlQueryRequest request, EsqlExecutionInfo executionInfo, EsqlStatement statement) {
+        String projectRouting = statement.setting(QuerySettings.PROJECT_ROUTING);
+        if (projectRouting == null) {
+            projectRouting = request.projectRouting();
+        }
+        if (projectRouting != null && executionInfo.isCrossClusterSearch() == false) {
```

Comment:
`executionInfo.isCrossClusterSearch() == false` is misleading.

We are populating project routing before FC call (as intended), but `executionInfo.clusterInfo` that is backing this condition is populated after. I believe it is only passing because `executionInfo.isCrossClusterSearch()` happens to be `false` for empty/non-initialized data structure.